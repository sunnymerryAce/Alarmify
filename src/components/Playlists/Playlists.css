.playlist-item {
  width: 100%;
}
.playlist-item img {
  width: 100%;
}

ul {
  list-style: none;
  padding: 0;
}
