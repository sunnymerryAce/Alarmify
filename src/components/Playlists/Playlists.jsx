import Swiper from 'swiper';
import React, { Component } from 'react';
import styled from 'styled-components';

const Playlist = styled.ul`
  list-style: none;
  padding: 0;
`;
const PlaylistItem = styled.li`
  width: 50%;
  background: transparent;
  img {
    width: 100%;
  }
`;
export default class Playlists extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isPlaying: false,
      playlists: [],
    };
  }
  componentDidMount() {
    if (document.querySelector('.swiper-slide')) {
      this.coverflow = new Swiper('.swiper-container', {
        init: false,
        effect: 'coverflow',
        grabCursor: true,
        loop: false,
        centeredSlides: true,
        slidesPerView: 'auto',
        coverflowEffect: {
          rotate: 50,
          stretch: 0,
          depth: 100,
          modifier: 1,
          slideShadows: true,
        },
        speed: 500,
      });
      this.coverflow.on('init', () => {
        this.props.onChangePlaylist(this.coverflow.activeIndex);
      });
      this.coverflow.on('slideChange', () => {
        this.props.onChangePlaylist(this.coverflow.activeIndex);
      });
      this.coverflow.init();
    }
  }

  render() {
    return (
      <div className="swiper-container">
        <Playlist className="swiper-wrapper">
          {this.props.playlists.map((item, key) => {
            return (
              <PlaylistItem key={key} className="swiper-slide">
                <img src={item.images[0].url} alt="" />
              </PlaylistItem>
            );
          })}
        </Playlist>
      </div>
    );
  }
}
