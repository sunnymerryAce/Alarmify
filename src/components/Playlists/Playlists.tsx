import Swiper from 'swiper';
import React, { useState, useEffect } from 'react';
import styled from 'styled-components';

interface Props {
  onChangePlaylist(index: number): void;
  playlists: Array<any>;
}
const Playlists: React.FC<Props> = props => {
  const [isPlaying, setIsPlaying] = useState<boolean>(false);

  useEffect(() => {
    console.log('useeffect');
    if (document.querySelector('.swiper-slide')) {
      const coverflow = new Swiper('.swiper-container', {
        effect: 'coverflow',
        grabCursor: true,
        loop: false,
        centeredSlides: true,
        slidesPerView: 'auto',
        coverflowEffect: {
          rotate: 50,
          stretch: 0,
          depth: 100,
          modifier: 1,
          slideShadows: true,
        },
        speed: 500,
      });
      if (coverflow) {
        props.onChangePlaylist(coverflow.activeIndex);
      }
      coverflow.on('slideChange', () => {
        if (coverflow) {
          props.onChangePlaylist(coverflow.activeIndex);
        }
      });
    }
  }, []);

  return (
    <div className="swiper-container">
      <Playlist className="swiper-wrapper">
        {props.playlists.map((item, key) => {
          return (
            <PlaylistItem key={key} className="swiper-slide">
              <img src={item.images[0].url} alt="" />
            </PlaylistItem>
          );
        })}
      </Playlist>
    </div>
  );
};

export default Playlists;
// export default class Playlists extends React.Component<Props, State> {
//   coverflow: Swiper | null;
//   constructor(props: Props) {
//     super(props);
//     this.state = {
//       isPlaying: false,
//     };
//     this.coverflow = null;
//   }
//   componentDidMount() {}

//   render() {
//     return (
//       <div className="swiper-container">
//         <Playlist className="swiper-wrapper">
//           {this.props.playlists.map((item, key) => {
//             return (
//               <PlaylistItem key={key} className="swiper-slide">
//                 <img src={item.images[0].url} alt="" />
//               </PlaylistItem>
//             );
//           })}
//         </Playlist>
//       </div>
//     );
//   }
// }

const Playlist = styled.ul`
  list-style: none;
  padding: 0;
`;
const PlaylistItem = styled.li`
  width: 50%;
  background: transparent;
  img {
    width: 100%;
  }
`;
