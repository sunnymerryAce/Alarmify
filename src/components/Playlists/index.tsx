import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import Swiper from 'swiper';

interface Props {
  playlists: Array<SpotifyApi.PlaylistObjectSimplified>;
  setPlaylistUri: React.Dispatch<React.SetStateAction<string>>;
}

const Playlists: React.FC<Props> = (props) => {
  const [coverFlow, setCoverFlow] = useState<Swiper>();
  const [currentIndex, setCurrentIndex] = useState<number>(0);

  useEffect(() => {
    initialize();
  }, []);

  // スライダーレンダリング後
  useEffect(() => {
    if (!coverFlow) return;
    setCurrentPlaylist(coverFlow.activeIndex);
    addCoverFlowEvent(coverFlow);
  }, [coverFlow]);

  const initialize = (): void => {
    setCoverFlow(createCoverFlow());
  };

  const createCoverFlow = (): Swiper => {
    return new Swiper('.swiper-container', {
      effect: 'coverflow',
      grabCursor: true,
      loop: false,
      centeredSlides: true,
      slidesPerView: 'auto',
      coverflowEffect: {
        rotate: 50,
        stretch: 0,
        depth: 100,
        modifier: 1,
        slideShadows: true,
      },
      speed: 500,
    });
  };

  const addCoverFlowEvent = (coverFlow: Swiper): void => {
    coverFlow.on('slideChange', () => {
      setCurrentPlaylist(coverFlow.activeIndex);
    });
  };

  const setCurrentPlaylist = (index: number): void => {
    setCurrentIndex(index);
    props.setPlaylistUri(props.playlists[index].uri);
  };

  return (
    <>
      <Title>{props.playlists[currentIndex].name}</Title>
      <div className="swiper-container">
        <CoverFlow className="swiper-wrapper">
          {props.playlists.map((item, key) => {
            return (
              <CoverFlowItem key={key} className="swiper-slide">
                <img
                  src={item.images.length ? item.images[0].url : ''}
                  alt=""
                />
              </CoverFlowItem>
            );
          })}
        </CoverFlow>
      </div>
    </>
  );
};

export default Playlists;

const Title = styled.div`
  font-size: 25px;
  font-weight: bold;
  letter-spacing: 0;
  padding-bottom: 2vh;
`;
const CoverFlow = styled.ul`
  list-style: none;
  padding: 0;
`;
const CoverFlowItem = styled.li`
  width: 50%;
  background: transparent;
  img {
    width: 100%;
  }
`;
