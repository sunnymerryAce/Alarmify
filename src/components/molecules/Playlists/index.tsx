import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import Swiper from 'swiper';

export const createCoverFlow = (): Swiper => {
  return new Swiper('.swiper-container', {
    effect: 'coverflow',
    grabCursor: true,
    loop: false,
    centeredSlides: true,
    slidesPerView: 'auto',
    coverflowEffect: {
      rotate: 50,
      stretch: 0,
      depth: 100,
      modifier: 1,
      slideShadows: true,
    },
    speed: 500,
  });
};

interface Props {
  playlists: Array<SpotifyApi.PlaylistObjectSimplified>;
  onChangePlaylist: (index: number) => void;
}

const Playlists: React.FC<Props> = (props) => {
  const [currentIndex, setCurrentIndex] = useState<number>(0);

  useEffect(() => {
    initialize();
    // 初期処理のため第二引数を指定しない
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const initialize = () => {
    // coverFlow生成
    const coverFlow = createCoverFlow();
    setCurrentIndex(coverFlow.activeIndex);
    props.onChangePlaylist(coverFlow.activeIndex);
    coverFlow.on('slideChange', () => {
      setCurrentIndex(coverFlow.activeIndex);
      props.onChangePlaylist(coverFlow.activeIndex);
    });
  };

  return (
    <>
      <Title className="playlist-title">
        {props.playlists[currentIndex]
          ? props.playlists[currentIndex].name
          : ''}
      </Title>
      <div className="swiper-container">
        <CoverFlow className="swiper-wrapper">
          {props.playlists.map((item, key) => {
            return (
              <CoverFlowItem key={key} className="swiper-slide">
                <img
                  src={item.images.length ? item.images[0].url : ''}
                  alt=""
                />
              </CoverFlowItem>
            );
          })}
        </CoverFlow>
      </div>
    </>
  );
};

export default Playlists;

const Title = styled.div`
  font-size: 25px;
  font-weight: bold;
  letter-spacing: 0;
  padding-bottom: 2vh;
`;
const CoverFlow = styled.ul`
  list-style: none;
  padding: 0;
`;
const CoverFlowItem = styled.li`
  width: 50%;
  background: transparent;
  img {
    width: 100%;
  }
`;
