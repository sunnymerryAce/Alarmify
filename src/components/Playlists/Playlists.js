import Swiper from 'swiper';
import React, { Component } from 'react';
import './Playlists.css';

export default class Playlists extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isPlaying: false,
      playlists: [],
    };
  }
  componentDidMount() {
    if (document.querySelector('.swiper-slide')) {
      this.coverflow = new Swiper('.swiper-container', {
        init: false,
        effect: 'coverflow',
        grabCursor: true,
        loop: false,
        centeredSlides: true,
        slidesPerView: 'auto',
        coverflowEffect: {
          rotate: 50,
          stretch: 0,
          depth: 100,
          modifier: 1,
          slideShadows: true,
        },
        speed: 500,
      });
      this.coverflow.on('init', () => {
        this.props.onChangePlaylist(this.coverflow.activeIndex);
      });
      this.coverflow.on('slideChange', () => {
        this.props.onChangePlaylist(this.coverflow.activeIndex);
      });
      this.coverflow.init();
    }
  }

  fetchPlayLists() {
    fetch(
      'https://asia-northeast1-alarmify-5f826.cloudfunctions.net/getPlaylists',
    )
      .then((response) => {
        return response.json();
      })
      .then((json) => {
        this.setState({
          isPlaying: this.state.isPlaying,
          playlists: json.items,
        });
      });
  }

  render() {
    return (
      <div className="Playlists swiper-container">
        <ul className="swiper-wrapper">
          {this.props.playlists.map((item, key) => {
            return (
              <li key={key} className="playlist-item swiper-slide">
                <img src={item.images[0].url} alt="" />
              </li>
            );
          })}
        </ul>
      </div>
    );
  }
}
