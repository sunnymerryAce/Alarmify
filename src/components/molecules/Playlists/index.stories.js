import React from 'react';
import Playlists from './index';

import 'modern-normalize/modern-normalize.css';
import 'swiper/css/swiper.min.css';
import '../../../App.css';

export default {
  title: 'Playlists',
};

export const Default = () => {
  const playlists = [
    {
      images: [
        {
          url:
            'https://mosaic.scdn.co/640/ab67616d0000b27321d2f32ee49927dd948f5a7aab67616d0000b2734a5389bfed87c750d57c6879ab67616d0000b2735a410ed481f7e851c2e99a6bab67616d0000b273cfe0ea65058f2584a17bca19',
        },
      ],
      name: 'About Time',
    },
    {
      images: [
        {
          url:
            'https://mosaic.scdn.co/640/ab67616d0000b27321d2f32ee49927dd948f5a7aab67616d0000b2734a5389bfed87c750d57c6879ab67616d0000b2735a410ed481f7e851c2e99a6bab67616d0000b273cfe0ea65058f2584a17bca19',
        },
      ],
      name: 'About Time',
    },
    {
      images: [
        {
          url:
            'https://mosaic.scdn.co/640/ab67616d0000b27321d2f32ee49927dd948f5a7aab67616d0000b2734a5389bfed87c750d57c6879ab67616d0000b2735a410ed481f7e851c2e99a6bab67616d0000b273cfe0ea65058f2584a17bca19',
        },
      ],
      name: 'About Time',
    },
  ];
  return <Playlists playlists={playlists} onChangePlaylist={() => {}} />;
};

export const Single = () => {
  const playlists = [
    {
      images: [
        {
          url:
            'https://mosaic.scdn.co/640/ab67616d0000b27321d2f32ee49927dd948f5a7aab67616d0000b2734a5389bfed87c750d57c6879ab67616d0000b2735a410ed481f7e851c2e99a6bab67616d0000b273cfe0ea65058f2584a17bca19',
        },
      ],
      name: 'About Time',
    },
  ];
  return <Playlists playlists={playlists} onChangePlaylist={() => {}} />;
};
