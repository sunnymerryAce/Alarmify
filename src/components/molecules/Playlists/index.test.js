import React from 'react';
import Swiper from 'swiper';
import Playlists, { createCoverFlow } from './index';
import { shallow } from 'enzyme';

export function createStartTouchEventObject({ x = 0, y = 0 }) {
  return { touches: [{ clientX: x, clientY: y }] };
}

export function createMoveTouchEventObject({ x = 0, y = 0 }) {
  return { changedTouches: [{ clientX: x, clientY: y }] };
}

describe('createCoverFlow', () => {
  const coverFlow = createCoverFlow();
  expect(coverFlow instanceof Swiper).toBe(true);
});

describe('<Playlists />', () => {
  const testImageUrl = 'test.png';
  const testTitle = 'title';
  const playlists = [
    {
      href: '',
      id: '',
      images: [
        {
          url: testImageUrl,
        },
      ],
      name: testTitle,
      snapshot_id: '',
      type: 'playlist',
      uri: '',
    },
    {
      href: '',
      id: '',
      images: [],
      name: '',
      snapshot_id: '',
      type: 'playlist',
      uri: '',
    },
  ];
  const onChangePlaylist = (number) => {
    console.log(number);
  };
  const container = shallow(
    <Playlists playlists={playlists} onChangePlaylist={onChangePlaylist} />,
  );

  test('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  test('should have proper components', () => {
    expect(container.find('.playlist-title').length).toBe(1);
    expect(container.find('.swiper-container').length).toBe(1);
  });

  test('should have proper title in initial state', () => {
    expect(container.find('.playlist-title').text()).toBe(testTitle);
  });

  test('should have proper length of slides', () => {
    expect(container.find('.swiper-slide').length).toBe(2);
  });

  test('should have proper image src', () => {
    expect(
      container
        .find('.swiper-slide')
        .at(0)
        .find('img')
        .props(),
    ).toMatchObject({
      src: testImageUrl,
    });
    expect(
      container
        .find('.swiper-slide')
        .at(1)
        .find('img')
        .props(),
    ).toMatchObject({
      src: '',
    });
  });
});
