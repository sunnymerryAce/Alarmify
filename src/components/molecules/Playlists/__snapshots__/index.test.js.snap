// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Playlists /> should match the snapshot 1`] = `"<div class=\\"sc-bdVaJa dZEyXr playlist-title\\">title</div><div class=\\"swiper-container\\"><ul class=\\"sc-bwzfXH iNIpfV swiper-wrapper\\"><li class=\\"sc-htpNat gFkvLr swiper-slide\\"><img src=\\"test.png\\" alt=\\"\\"/></li><li class=\\"sc-htpNat gFkvLr swiper-slide\\"><img src=\\"\\" alt=\\"\\"/></li></ul></div>"`;
